### PR TITLE
Integrate MindSpore into HF Inference API

### PR DESCRIPTION
**Pull request by hellowaywewe**
MindSpore is a new open-source deep learning training/inference framework that could be used for mobile, edge, and cloud scenarios. To make the widgets on the hub work, we use TinyMS, MindSopore's easy-to-use suite library, to adapt to HF to provide more AI experience services for users.

### CODE REVIEW

File: docker_images/mindspore/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+FROM tiangolo/uvicorn-gunicorn:python3.8
+LABEL maintainer="me <me@example.com>"
+
+# Add any system dependency here
+RUN apt-get update -y && apt-get install libglib2.0-dev libsm6 libxrender1 libgl1-mesa-glx -y
+
+COPY requirements.txt /app
+RUN /usr/local/bin/python -m pip install --upgrade pip && \
+    pip install --no-cache-dir -r requirements.txt && \
+    pip install --no-cache-dir git+https://github.com/tinyms-ai/tinyms
```
Comment (Narsil) [inline docker_images/mindspore/Dockerfile:10]: Can we move this to `requirements.txt`  ? it does work on github installs too.

**Review by Narsil — COMMENTED** [review]
Thanks for this PR it looks in great shape.

I might add a few files to get CI/CD more complete so that merges would happen automatically on merge.

File: docker_images/mindspore/app/pipelines/test_hf.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+def load_config(config_json_file):
+    with open(config_json_file, "r", encoding="utf-8") as reader:
+        config = reader.read()
+    return json.loads(config)
+
+
+config_json_file = os.path.join(repo_path, "config.json")
+config = load_config(config_json_file)
+print(config.get('architecture'))
```
[the remark this answers is not in the document]
Comment (hellowaywewe) [reply docker_images/mindspore/app/pipelines/test_hf.py:20]: @Narsil, Sorry for that, I mistakenly submit the local self-test file, and I will delete it later. Thank you for your quick reply to my pull request.

File: docker_images/mindspore/Dockerfile (hunk 1 above)
Comment (hellowaywewe) [reply docker_images/mindspore/Dockerfile:10]: Yes, I will update it.

**Comment by hellowaywewe** [comment]
@Narsil Thank you for your review, I will update it.

File: docker_images/mindspore/app/pipelines/test_hf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Narsil) [reply docker_images/mindspore/app/pipelines/test_hf.py:20]: No worries !

**Comment by hellowaywewe** [comment]
@Narsil @xianbaoqian, I update this PR, please review it again, thanks.

**Review by Narsil — APPROVED** [review]
LGTM. 

I added the GH actions and fixed the style for you (`make style`) !

**Comment by xianbaoqian** [comment]
Thank you! Looks great! Please also be aware of the changes required in https://github.com/huggingface/hub-docs also :-)

**Review by xianbaoqian — APPROVED** [review]

**Comment by Narsil** [comment]
Operational:

`curl -X POST --data-binary "@shard/tests/plane.jpg" https://api-inference.huggingface.co/models/mindspore-ai/LeNet -H "x-wait-for-model: 1"`

**Pull request merged**